Building on OSX 10.10

Just tried to build on Mac OS X 10.10 and always getting this:

CMake Error at /usr/local/Cellar/cmake/3.0.0/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:136 (message):
  Could NOT find PythonLibs: Found unsuitable version "2.7.6", but required
  is at least "3.3" (found
  /usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/libpython3.4.dylib)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.0.0/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:341 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.0.0/share/cmake/Modules/FindPythonLibs.cmake:276 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cpp/CMakeLists.txt:74 (find_package)

The cmake from brew (3.0.2) didn't work. I've installed 3.0.0 from https://raw.githubusercontent.com/NikolausDemmel/homebrew-devel/master/cmake.rb (with some sha1 manipulation), still no results.

Will be thankful for any support, and will be glad to help with Mac testings.

--- a/buildsystem/modules/FindPython.cmake
+++ b/buildsystem/modules/FindPython.cmake
@@ -1,0 +1,85 @@
+# Find Python
+# ~~~~~~~~~~~
+# Find the Python interpreter and related Python directories.
+#
+# This file defines the following variables:
+#
+# PYTHON_FOUND       - True when python was found
+#
+# PYTHON_EXECUTABLE  - The full path to the Python interpreter.
+#
+# PYTHON_INCLUDE_DIR - Directory holding the include files.
+#
+# PYTHON_LIBRARY     - Location of the Python library.
+
+function(find_frameworks list framework_name)
+	set(_hints
+			/usr/local/Frameworks
+			~/Library/Frameworks
+			/System/Library/Frameworks
+	)
+	foreach(_hint ${_hints})
+		set(_current_folder "${_hint}/${framework_name}.framework")
+		if(IS_DIRECTORY ${_current_folder})
+			SET(_list ${_list} ${_current_folder})
+		endif()
+	endforeach()
+	set(${list} ${_list} PARENT_SCOPE)
+endfunction()
+
+function(find_python_framework framework_paths)
+	find_frameworks(PYTHON_FRAMEWORKS "Python")
+	foreach(_framework_path ${PYTHON_FRAMEWORKS})
+		file(GLOB children RELATIVE ${_framework_path}/Versions ${_framework_path}/Versions/*)
+		foreach(child ${children})
+			if(IS_DIRECTORY ${_framework_path}/Versions/${child})
+				if((${Python_FIND_VERSION} VERSION_EQUAL ${child}) OR ((NOT Python_FIND_VERSION_EXACT) AND (NOT ${Python_FIND_VERSION} VERSION_GREATER ${child})))
+					SET(_matching_framework_paths ${_matching_framework_paths} "${_framework_path}/Versions/${child}")
+				endif()
+			endif()
+		endforeach()
+	endforeach()
+	set(${framework_paths} ${_matching_framework_paths} PARENT_SCOPE)
+endfunction()
+
+if (EXISTS "${PYTHON_EXECUTABLE}" AND EXISTS "${PYTHON_INCLUDE_DIR}" AND EXISTS "${PYTHON_LIBRARY}")
+	set(PYTHON_FOUND TRUE)
+endif()
+if(APPLE AND NOT PYTHON_FOUND)
+	find_python_framework(PYTHON_FRAMEWORK_PATHS)
+	if(PYTHON_FRAMEWORK_PATHS)
+		list(GET PYTHON_FRAMEWORK_PATHS 0 PYTHON_FRAMEWORK)
+		get_filename_component(PYTHON_VERSION "${PYTHON_FRAMEWORK}" NAME)
+		set(PYTHON_EXECUTABLE "${PYTHON_FRAMEWORK}/bin/python${PYTHON_VERSION}")
+		set(PYTHON_INCLUDE_DIR "${PYTHON_FRAMEWORK}/Headers")
+		find_library(PYTHON_LIBRARY "python${PYTHON_VERSION}" PATHS "${PYTHON_FRAMEWORK}/lib")
+		if(EXISTS "${PYTHON_EXECUTABLE}" AND EXISTS "${PYTHON_INCLUDE_DIR}" AND EXISTS "${PYTHON_LIBRARY}")
+			set(PYTHON_FOUND TRUE)
+		endif()
+	endif()
+endif()
+if(NOT PYTHON_FOUND)
+	set(Python_ADDITIONAL_VERSIONS 3.4)
+	# find py libs
+	find_package(PythonLibs 3.3 REQUIRED)
+	# on some distros (ubuntu 13.10) there seems to be additional clutter at the end,
+	# which needs to be stripped to prevent find_package(PythonInterp) from erroring.
+	if(PYTHONLIBS_VERSION_STRING MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)([^0-9].*)?$")
+		set(PYTHON_SANITIZED_VERSION "${CMAKE_MATCH_1}")
+	else()
+		message(WARNING "PYTHONLIBS_VERSION_STRING is ${PYTHONLIBS_VERSION_STRING}, expected major.minor.patch*")
+		set(PYTHON_SANITIZED_VERSION "${PYTHONLIBS_VERSION_STRING}")
+	endif()
+	# look for the py interpreter that fits to the py libs
+	find_package(PythonInterp "${PYTHON_SANITIZED_VERSION}" EXACT REQUIRED)
+	if(PYTHONLIBS_FOUND AND PYTHONINTERP_FOUND)
+		set(PYTHON_FOUND TRUE)
+	endif()
+endif()
+if(Python_FIND_REQUIRED AND NOT PYTHON_FOUND)
+	if(Python_FIND_VERSION_EXACT)
+		message(FATAL_ERROR "Could not find Python exactly of version ${Python_FIND_VERSION}.")
+	else()
+		message(FATAL_ERROR "Could not find Python of version ${Python_FIND_VERSION} or greater.")
+	endif()
+endif()

--- a/buildsystem/python.cmake
+++ b/buildsystem/python.cmake
@@ -5,22 +5,7 @@
 # for an example usage, see py/openage/convert/CMakeLists.txt
 
 function(python_init)
-	# cmake 2.8 compatibility
-	set(Python_ADDITIONAL_VERSIONS 3.4)
-	# find py libs
-	find_package(PythonLibs 3.3 REQUIRED)
-
-	# on some distros (ubuntu 13.10) there seems to be additional clutter at the end,
-	# which needs to be stripped to prevent find_package(PythonInterp) from erroring.
-	if(PYTHONLIBS_VERSION_STRING MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)([^0-9].*)?$")
-		set(PYTHON_SANITIZED_VERSION "${CMAKE_MATCH_1}")
-	else()
-		message(WARNING "PYTHONLIBS_VERSION_STRING is ${PYTHONLIBS_VERSION_STRING}, expected major.minor.patch*")
-		set(PYTHON_SANITIZED_VERSION "${PYTHONLIBS_VERSION_STRING}")
-	endif()
-
-	# look for the py interpreter that fits to the py libs
-	find_package(PythonInterp "${PYTHON_SANITIZED_VERSION}" EXACT REQUIRED)
+	find_package(Python 3.3 REQUIRED)
 
 	# these following lists are filled by the add_py_package and add_pyext_module functions:
 	set_property(GLOBAL PROPERTY "SFT_PY_PACKAGES")

